### PR TITLE
Allow customizing the key property type

### PR DESCRIPTION
**Pull request by danielpza**
Why? This fails:

```tsx
const [selected, setSelected] = new Set<string>([])

<DataGrid selectedRows={selected} onSelectedRowsChange={setSelected} />
```

because onSelectedRowsChange was expected to receive all the properties React.Key has, eg number, which is not possible/desired.

### CODE REVIEW

File: src/DataGrid.tsx [hunk 1]
```diff
@@ -80,7 +80,7 @@ type SharedDivProps = Pick<React.HTMLAttributes<HTMLDivElement>,
   | 'style'
 >;
 
-export interface DataGridProps<R, SR = unknown> extends SharedDivProps {
+export interface DataGridProps<R, SR = unknown, K = React.Key> extends SharedDivProps {
```
Comment (nstepien) [inline src/DataGrid.tsx:83]:
```suggestion
export interface DataGridProps<R, SR = unknown, K extends React.Key = React.Key> extends SharedDivProps {
```

**Review by nstepien — CHANGES_REQUESTED** [review]
I think it's a good idea.
Can you please also add the generic to the default export, it's at the end of `DataGrid.tsx`.

**Comment by nstepien** [comment]
Please also add the generic to the default export at the end of `DataGrid.tsx`, it won't be usable otherwise.

File: src/DataGrid.tsx [hunk 2]
```diff
@@ -1071,4 +1071,4 @@ function DataGrid<R, SR>({
   );
 }
 
-export default forwardRef(DataGrid) as <R, SR = unknown>(props: DataGridProps<R, SR> & RefAttributes<DataGridHandle>) => JSX.Element;
+export default forwardRef(DataGrid) as <R, SR = unknown, Key extends React.ReactKey = React.ReactKey>(props: DataGridProps<R, SR, Key> & RefAttributes<DataGridHandle>) => JSX.Element;
```
Comment (nstepien) [inline src/DataGrid.tsx:1074]:
```suggestion
export default forwardRef(DataGrid) as <R, SR = unknown, Key extends React.Key = React.Key>(props: DataGridProps<R, SR, Key> & RefAttributes<DataGridHandle>) => JSX.Element;
```

**Comment by danielpza** [comment]
Thank you! :tada:

**Pull request merged**